pkg/operators/otel-metrics/otel-metrics.go

```diff
@@ -100,16 +104,37 @@ func (m *otelMetricsOperator) Init(globalParams *params.Params) error {
 		go func() {
 			mux := http.NewServeMux()
 			mux.Handle("/metrics", promhttp.Handler())
-			err := http.ListenAndServe(globalParams.Get(ParamOtelMetricsListenAddress).AsString(), mux)
-			if err != nil {
+
+			m.server = &http.Server{
+				Addr:    globalParams.Get(ParamOtelMetricsListenAddress).AsString(),
+				Handler: mux,
+			}
+
+			// ErrServerClosed on graceful close
+			err := m.server.ListenAndServe()
+			if err != nil && !errors.Is(err, http.ErrServerClosed) {
 				log.Errorf("serving otel metrics on: %s", err)
+				m.server = nil
 				return
 			}
 		}()
 	}
 	return nil
 }
 
+func (m *otelMetricsOperator) Close() {
```

Comment:
Shouldn't we call these `Close` functions on [ops](https://github.com/inspektor-gadget/inspektor-gadget/blob/jose/test-otel-init/cmd/common/oci.go#L276)?

Reply:
But it should be handled by the gadget-context, shouldn't it be?

Reply:
No - this is regarding the global operator part, not the operator instance (which is handled by the gadget-context).